### PR TITLE
Add unit tests

### PR DESCRIPTION
**Pull request by sameeragunarathne**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: component/pom.xml [hunk 1]
```diff
@@ -86,6 +86,120 @@
...
+        </dependency>
+        <dependency>
+            <groupId>log4j</groupId>
+            <artifactId>log4j</artifactId>
+            <version>${log4j.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.powermock</groupId>
+            <artifactId>powermock-module-junit4</artifactId>
+            <version>1.6.5</version>
```
Comment (Biruntha) [inline component/pom.xml]: Move version into properties

File: component/src/test/resources/testng.xml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd" >
+<!--
+  ~ Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (Biruntha) [inline component/src/test/resources/testng.xml]: Change license version

File: component/src/test/resources/testng.xml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  -->
+
+<suite name="Suite1" verbose="1" >
+
+    <test name="Office365-Authenticator-Unit-Tests">
+        <classes>
+            <class name="org.wso2.carbon.identity.authenticator.office365.Office365AuthenticatorTest"/>
+        </classes>
+    </test>
+</suite>
```
Comment (Biruntha) [inline component/src/test/resources/testng.xml]: Add new line

File: pom.xml [hunk 4]
```diff
@@ -333,5 +370,14 @@
         <encoder.wso2.version>1.2.0.wso2v1</encoder.wso2.version>
         <org.apache.oltu.oauth2.client>0.31</org.apache.oltu.oauth2.client>
         <org.apache.oltu.oauth2.common>1.0.1</org.apache.oltu.oauth2.common>
+        <log4j.version>1.2.13</log4j.version>
+        <!--Test Dependencies-->
+        <junit.version>4.12</junit.version>
+        <testng.version>6.9.10</testng.version>
+        <jacoco.version>0.7.9</jacoco.version>
+        <powermock.version>1.6.5</powermock.version>
+        <!--<slf4j.api.version>1.7.12</slf4j.api.version>-->
```
Comment (Biruntha) [inline pom.xml]: remove commented section

File: component/pom.xml [hunk 5]
```diff
@@ -86,6 +86,120 @@
...
+                        </goals>
+                        <configuration>
+                            <rules>
+                                <rule implementation="org.jacoco.maven.RuleConfiguration">
+                                    <element>BUNDLE</element>
+                                    <limits>
+                                        <limit implementation="org.jacoco.report.check.Limit">
+                                            <counter>COMPLEXITY</counter>
+                                            <value>COVEREDRATIO</value>
+                                            <!--<minimum>0.60</minimum>-->
```
Comment (Biruntha) [inline component/pom.xml]: remove commented section

**Comment by madurangasiriwardena** [comment]
@sameeragunarathne, can you please check the comments added and resolve the conflicts.

**Pull request closed without merging**